src/Security/InMemoryUserEmailAddressRepository.php

```diff
@@ -4,25 +4,20 @@
 
 namespace CultuurNet\UDB3\Security;
 
+use CultuurNet\UDB3\Http\Auth\Jwt\JsonWebToken;
 use CultuurNet\UDB3\Model\ValueObject\Web\EmailAddress;
 
 final class InMemoryUserEmailAddressRepository implements UserEmailAddressRepository
 {
-    /**
-     * @var EmailAddress[]
-     */
-    private static array $mappedUserIds = [];
+    private JsonWebToken $token;
 
-    public static function addUserEmail(string $userId, EmailAddress $emailAddress): void
+    public function __construct(JsonWebToken $token)
     {
-        self::$mappedUserIds[$userId] = $emailAddress;
+        $this->token = $token;
     }
 
     public function getEmailForUserId(string $userId): ?EmailAddress
     {
-        if (array_key_exists($userId, self::$mappedUserIds)) {
-            return self::$mappedUserIds[$userId];
-        }
-        return null;
+        return $this->token->getEmailAddress();
```

Comment:
We should probably double-check here that the given `string $userId` is the same as the user id inside the token, to avoid bugs down the line when/if someone uses this class in another place

Reply:
Good find, I've made the changes.